### ISSUE TITLE
Incorrect count results by COUNT DISTINCT with FILTER and GROUP BY query

### ISSUE THREAD

**Issue by t-yuki** [duckdb/duckdb#5070]
### What happens?

When I use `COUNT (DISTINCT ...) FILTER ...` columns on a query with `GROUP BY`, DuckDB returns `0` count results for second and more rows. 

Firstly I found this issue on my local db with large tables, but also it reproduces on a simple `VALUES` in memory table as "To Reproduce" section.

I've checked that the query results `(1, 1)` on other SQL systems (Presto and SQLite).

Also, I've checked the result of DuckDB v0.4.0 da9ee490d but it also returns `0` count result: `('1', 0)`.

Sorry for if it is the expected results for DuckDB.

### To Reproduce

Run:

```
WITH evs AS (
  SELECT * FROM (VALUES 
    ('1','123','7'),
    ('1','456','7')
  ) AS t("id", "type", "value" )
)
SELECT "id"
, COUNT(DISTINCT "value") FILTER (WHERE "type" = '456') AS type_456_count
FROM evs
GROUP BY "id"
```

Expected Results:

`[('1', 1)]`

Actual Results:

`[('1', 0)]`

Description:

`type_456_count` column should be `1` but the result is `0`.

### OS:

Windows 10 AMD64

### DuckDB Version:

0.5.1

### DuckDB Client:

v0.5.1 7c111322d

### Full Name:

Yukinari Toyota

### Affiliation:

None

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by t-yuki** [comment]
Note that, `APPROX_COUNT_DISTINCT` isn't affected by this issue. It returns `1` correctly.

Also, removing `('1','123','7')` row entirely from VALUES clause fixes the result. `type_456_count` column returns `1` correctly.

**Comment by Tishj** [comment]
Sounds like a bug, I'm going to be working on the grouped distinct aggregate case so I'll have a look at this issue!

Thanks for the bug report! :thumbsup:

**Comment by Tishj** [comment]
```sql
query II
WITH evs AS (
  SELECT * FROM (VALUES 
    ('1','123','7'),
    ('1','456','7')
  ) AS t("id", "type", "value" )
)
SELECT "id"
, COUNT(DISTINCT "value") FILTER (WHERE "type" = '456') AS type_456_count
FROM evs
GROUP BY "id"
----
1	1

```

```
===============================================================================
All tests passed (2 assertions in 1 test case)
```
👍 this is fixed by #5146
